### PR TITLE
CI: Using docstring validator from numpydoc

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] xref #28822
- [x] tests added / passed
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

We moved our script to numpydoc, and it already had some improvements there. The script does like 80% of our validation, so what I'm doing here is to call numpydoc validation, and then call our custom validation (things that for different reasons weren't moved to numpydoc).

### CODE REVIEW

File: scripts/validate_docstrings.py [hunk 1]
```diff
@@ -1021,7 +400,7 @@ def header(title, width=80, char="#"):
         choices=format_opts,
         help="format of the output when validating "
         "multiple docstrings (ignored when validating one)."
-        f"It can be {str(format_opts)[1:-1]}",
+        "It can be {}".format(str(format_opts)[1:-1]),
```
Comment (WillAyd) [inline scripts/validate_docstrings.py]: I think worth keeping the f-strings

**Review by WillAyd — CHANGES_REQUESTED** [review]
Makes sense. For the codes that are left over do you see another home for them, like code checks or any of the other validators? Would be nice to do away with these scripts altogether if mostly covered by third parties

File: scripts/validate_docstrings.py (hunk 1 above)
Comment (datapythonista) [reply scripts/validate_docstrings.py]: Yep, missed that in a merge, which are quite tricky. Thanks.

**Comment by datapythonista** [comment]
> For the codes that are left over do you see another home for them, like code checks or any of the other validators?

The assumption here is that we want to be able to have a script that reports all the errors for a specific docstring. The one we used in the sprints, which we'll be using again when fixing one docstring at a time.

Otherwise, we could do what you say. And the validation for pep8 errors in the examples, which is the most tricky one, could just be deleted, since we can use pytest for that.

One thing to consider is that the numpydoc script is expected to be wrapped if used in the CI. It can be used directly for a single object, but we need to implement the functionality to get which objects we want to validate, which I think it's more than half of the new script.

**Comment by datapythonista** [comment]
@WillAyd can you remove the requested changes flag if you don't have further comments. Thanks!

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by datapythonista** [comment]
Merging this today if there are no further comments.

**Pull request merged**